src/topicFactory.js

```diff
@@ -0,0 +1,93 @@
+import nets from 'nets'
+import qs from 'qs'
+import randomString from './util/randomString'
+const CHASQUI_URL = 'https://chasqui.uport.me/api/v1/topic/'
+
+function TopicFactory (isOnMobile, pollingInterval = 2000, chasquiUrl = CHASQUI_URL) {
+  function waitForHashChange (topicName, cb) {
```

Comment:
This is higher order function meaning a function that returns a function instead of an object. It's safer and less error prone. Also much easier to mock for tests.